absl/base/internal/unscaledcycleclock.cc

```diff
@@ -57,11 +62,34 @@ double UnscaledCycleClock::Frequency() {
 #elif defined(__powerpc__) || defined(__ppc__)
 
 int64_t UnscaledCycleClock::Now() {
+#ifdef __GLIBC__
   return __ppc_get_timebase();
+#elif defined(__FreeBSD__)
+  union { long long complete; unsigned int part[2]; } ticks;
```

Comment:
Though you sometimes see it, type-punning through a union is undefined behavior in C++. See https://adriann.github.io/undefined_behavior.html for example.